plover/formatting.py

```diff
@@ -156,16 +156,31 @@ def _actions_to_text(action_list, text=u''):
         for a in action_list:
             if a.replace and text.endswith(a.replace):
                 text = text[:-len(a.replace)]
+
+            # Annoyance from after spaces: with certain commands spaces are placed to maintain state.
+            if a.replace == a.text == a.space_char and text == '':
+                text = a.word + a.text
             # With numbers, it's possible to have a.text='2' with a.word='1.2'
-            # folowing by an action that replaces '1.2' by '$1.20'...
-            if len(a.word) > len(a.text) and a.word.endswith(text):
-                text = a.word
+            # following by an action that replaces '1.2' by '$1.20'...
+            elif (len(a.word) > len(a.text)
+                      and a.word.endswith(text)):
+                text = OutputHelper._squish_strings(a.word, a.text)
```

Comment:
I think using anything else than `a.replace` and `a.text` was a mistake, and for our sanity, we should rework the code to only use those. See WIP [here](https://github.com/openstenoproject/plover/compare/master...benoit-pierre:translation_with_more_state). Ideally, we should even assert if `text` does not end with `a.replace`.